scripts/winesapos-setup.sh

```diff
@@ -67,9 +67,28 @@ else
 fi
 echo "Turning on the Mac fan service if the hardware is Apple complete."
 
+# Dialog to ask the user what mirror region they want to use
+# Fetch the list of regions from the Arch Linux mirror status JSON API
+arch_mirror_regions=$(curl -s https://archlinux.org/mirrors/status/json/ | \
+                        grep -oP '(?<="country_code": ")[^"]*' | sort | uniq) # grep maybe fragile, woule be better if we can use jq
+chosen_region=$(kdialog --title "winesapOS First-Time Setup" \
+                        --combobox "Select your desired mirror region, \nor press Cancel to use the Arch worldwide mirror:" ${arch_mirror_regions})
+
+# Append the region to /etc/xdg/reflector/reflector.conf (--country C1,C2,C3) if a region was chosen
+if [ -n "${chosen_region}" ]; then
+    echo "Chosen region: ${chosen_region}"
+    echo "--country ${chosen_region}" >> /etc/xdg/reflector/reflector.conf
```

Comment:
This does not work since the setup script is run as a non-privileged user. You'll have to do similar logic to what you do later by using a combination of `echo` and `sudo tee`. This should work to append to the file:

```
    echo "--country ${chosen_region}" | sudo tee -a /etc/xdg/reflector/reflector.conf
```

Reply:
Oops... didn't notice that the entire script isn't ran as root